src/mango_idx_text.erl

```diff
@@ -283,6 +283,23 @@ indexable_fields(Fields, {op_and, Args}) when is_list(Args) ->
     lists:foldl(fun(Arg, Fields0) -> indexable_fields(Fields0, Arg) end,
         Fields, Args);
 
+%% For queries that use array element access or $in operations, two
+%% fields get generated by mango_selector_text:convert. At index
+%% definition time, only one field gets defined. In this situation, we
+%% remove the extra generated field so that the index can be used. For
+%% all other situations, we include the fields as normal.
+indexable_fields(Fields, {op_or, [{op_field, Field0},
+        {op_field, {[Name | _], _}} = Field1]}) ->
+    case lists:member(<<"[]">>, Name) of
+        true ->
+            indexable_fields(Fields, Field1);
+        false ->
+            Fields1 = indexable_fields(Fields, Field0),
+            indexable_fields(Fields1, Field1)
+    end;
+indexable_fields(Fields, {op_or, Args}) when is_list(Args) ->
```

Comment:
@kxepal : The first function clause addresses two unique situations.

First is when the user does an $in operation with a list. Each element in the list will construct two fields with an op_or:
https://github.com/apache/couchdb-mango/blob/master/src/mango_selector_text.erl#L241-L260

The second situation is when the user access individual elements in the list. Again, there are two fields created with an op_or:
https://github.com/apache/couchdb-mango/blob/master/src/mango_selector_text.erl#L184-L191

In both these situations, the second op_field will contain a "[]" reference, which is a special identifier for that says we are referencing an array. So you may ask? What if a user asks

Give me all documents where the favorites field is equal to 5 or the first element a favorites array is 10.

"$or": [{"favorites": 5}, {"favorites.0": 10}]

This actually creates a nested op_or and will bypass the first function clause. Then the first "favorites": 5 will just be an op_field, while the second   {"favorites.0": 10} will be create an op_or that will be handled by the first function clause. 

So this way, we fix $in, and array element access without breaking $or.